pyfar/dsp/dsp.py

```diff
@@ -2669,3 +2669,173 @@ def normalize(signal, reference_method='max', domain='auto',
         return normalized_signal, reference_norm
     else:
         return normalized_signal
+
+
+def correlate(signal_1, signal_2, mode='linear'):
+    r"""
+    Compute channel-wise correlation between signals.
+
+    The correlation between the time signals :math:`x_1[n]` and :math:`x_2[n]`
+    is given by
+
+    .. math:: c[l] = \sum_n x_1[n] x_2[n-l]
+
+    with the index :math:`n` and time lag :math:`l` in samples. The computation
+    is realized in the frequency domain using the corresponding spectra
+    :math:`X_1[k]` and :math:`X_2[k]`
+
+    .. math:: c = \mathrm{IFFT\{X_1 X_2^-\}}
+
+    with the inverse fourier transform denoted by IFFT and :math:`X^-` denoting
+    the spectrum of a time-reversed and complex conjugated time signal, i.e.,
+    ``X_minus = fft(conj(x[::-1]))`` where the conjugate is required if the
+    input has complex-valued time data.
+
+    Parameters
+    ----------
+    signal_1 : Signal
+        The first input signal. It must have the same sample rate as
+        `signal_2` and its ``cshape`` must be broadcastable to that of
+        `signal_2`.
+    signal_2 : Signal
+        The second input signal. It must have the same sample rate as
+        `signal_1` and its ``cshape`` must be broadcastable to that of
+        `signal_1`.
+    mode : str, optional
+
+        ``'linear'``
+            Computes the linear correlation by zero padding the input signals
+            to a length of ``signal_1.n_samples + signal_2.n_samples - 1``
+            before applying the Fourier transform (see equations above).
+        ``'cyclic'``
+            Computes the cyclic correlation, which uses the input signals as
+            they are. In this case `signal_1` and `signal_2` must have the
+            same number of samples (length).
+
+        The default is ``'linear'``.
```

Comment:
The mode parameters are not in agreement with the pyfars convolution function:
https://pyfar.readthedocs.io/en/stable/modules/pyfar.dsp.html#pyfar.dsp.convolve

Reply:
Good point - changed it to 'full' everywhere.